torch/csrc/distributed/rpc/rref_context.cpp

```diff
@@ -16,7 +17,9 @@ void confirmPendingUser(
     const FutureMessage& futureMessage,
     const ForkId& expectedForkId) {
   if (!futureMessage.hasError()) {
-    auto rr = RemoteRet::fromMessage(futureMessage.constValue());
+    auto msgType = futureMessage.constValue().type();
+    auto rr = dynamic_cast<RemoteRet*>(
```

Comment:
This is casting `unique_ptr` to raw pointer? Does this guarantee the unique_ptr won't be destructed after this line?

Reply:
It looks like the recommended approach here is something like:
```
std::unique_ptr<RemoteRet> rr(dynamic_cast<RemoteRet*>(
        deserializeResponse(futureMessage.constValue(), msgType).release()));
```
 Also I don't like this runtime dynamic_cast here too much, but not sure if there's a way around it, since deserializeResponse returns a base type that RemoteRet inherits from.

Reply:
I see two options here, there might be more:

1. as all we need to is to keep the unique_ptr alive, we can have an additional `unique_ptr< RpcCommandBase>` tmp var. Sth like:
    ```c++
    auto rpc = deserializeResponse(futureMessage.constValue(), msgType);
    auto rr = dynamic_cast<RemoteRet*>(rpc.get());
    ```
2. Put everything into one line, IIUC, this will guarantee the temp var stay alive until the next `;`. Sth like:
    ```c++
    auto forkId = dynamic_cast<RemoteRet*>(deserializeResponse(futureMessage.constValue(), msgType).get())->forkId();
    ```

Reply:
Sounds good. Will go with the first, as I think it's a bit easier to read than putting everything in one line.